602-spring-data-rest/src/test/java/org/acme/spring/data/rest/BookRepositoryTest.java

```diff
@@ -0,0 +1,160 @@
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import org.acme.spring.data.rest.containers.PostgreSqlDatabaseTestResource;
+import org.junit.Assert;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.core.IsNot.not;
+
+@QuarkusTest
+@QuarkusTestResource(PostgreSqlDatabaseTestResource.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class BookRepositoryTest {
+    @Inject
+    BookRepository bookRepository;
+
+    @Inject
+    LibraryRepository libraryRepository;
+
+    @Test
+    @Order(1)
+    void testListAllBooks() throws InterruptedException {
+        //List all books (should have all 4 books the database has initially)
+        given()
+                .accept("application/json")
+                .when().get("/books")
+                .then()
+                .statusCode(200)
+                .body(
+                        containsString("Aeneid"),
+                        containsString("Beach House"),
+                        containsString("Cadillac Desert"),
+                        containsString("Dagon and Other Macabre Tales")
+                );
+
+        //Create a new Book
+        given()
+                .contentType("application/json")
+                .accept("application/json")
+                .body("{\"name\": \"Early Asimov\", \"author\": \"Isaac Asimov\"}")
+                .when().post("/books")
+                .then()
+                .statusCode(201)
+                .body(containsString("Early Asimov"))
+                .body("id", notNullValue())
+                .extract().body().jsonPath().getString("id");
+
+        //Find new book by id
+        given()
+                .when().get("/books/id/5")
+                .then()
+                .statusCode(200)
+                .body(
+                        containsString("Early Asimov")
+                );
+
+        //Try to delete a book via HTTP (method not allowed)
+        given()
+                .when().delete("/books/1")
+                .then()
+                .statusCode(405);
+
+        //Try to insert book with invalid constraints
+        given()
+                .contentType("application/json")
+                .body("{\"name\": \"Q\", \"author\": \"Li\"}")
+                .when().post("/books")
+                .then()
+                .statusCode(500) // Must be 400 as an error in client's data
+                .body(
+                        containsString("length must be between 2 and 50"),
```

Comment:
This is not passing for me locally as my default locale is Spanish. Can you adjust the property to set the default language to English always or add a custom message to the length property as follows:

```
@Length(min = 2, max = 50, message = "length must be between {min} and {max}")
```

Reply:
good catch